src/d/d_message.cpp

```diff
@@ -1,20 +1,1166 @@
+    return field_0x46;
+}
+
+bool dFlow_c::triggerEntryPointChecked(s32 labelPart1, s32 labelPart2) {
+    if (checkField0x3C()) {
+        s32 old = field_0x40;
+        dFlow_c::triggerEntryPoint(labelPart1, labelPart2);
+        setNext(old);
+        return true;
+    }
+    return false;
+}
+
+void dFlow_c::setField0x3C() {
+    field_0x3C = 1;
+}
+
+bool dFlow_c::checkField0x3C() const {
+    return field_0x3C == 1;
+}
+
+extern "C" dFlow_c *CURRENT_ACTOR_EVENT_FLOW_MANAGER;
+void dFlow_c::start(u16 entry) {
+    field_0x0E = 0;
+    field_0x0F = 0;
+    field_0x3C = 0;
+    field_0x40 = -1;
+    field_0x44 = -1;
+    field_0x46 = -1;
+    mDelayTimer = 0;
+    std::memset(&mFlowInfo, 0, sizeof(MsbFlowInfo));
+    CURRENT_ACTOR_EVENT_FLOW_MANAGER = this;
+    field_0x0C = 0xFFFF;
+    mCurrentFlowIndex = -1;
+    field_0x10 = 1;
+    setNext(entry);
+}
+
+bool dFlow_c::advanceUntil(s32 searchType, s32 searchParam3, s32 *pOutParams1n2) {
+    bool keepGoing = true;
+    while (keepGoing && !vt_0x18()) {
+        MsbFlowInfo *element = LMS_GetFlowElement(mpMsbf, mCurrentFlowIndex);
+        s32 type = element->type;
+        if (searchType == type) {
+            if (element->type == FLOW_EVENT) {
+                if (element->param3 == searchParam3) {
+                    if (pOutParams1n2 != nullptr) {
+                        *pOutParams1n2 = element->params1n2;
+                    }
+                    return true;
+                }
+            } else {
+                return true;
+            }
+        }
+        switch (type) {
+            case FLOW_MESSAGE:
+                // Skip text processing, simply advance
+                setNext(element->next);
+                keepGoing = true;
+                break;
+            case FLOW_BRANCH:
+                if (element->param3 <= 2) {
+                    keepGoing = false;
+                } else {
+                    keepGoing = handleBranch();
+                }
+                break;
+            case FLOW_EVENT:
+                switch (element->param3) {
+                    case 7:
+                    case 23:
+                    case 27: keepGoing = handleEvent(); continue;
+                    case 34:
+                        if (element->params1n2 == 2) {
+                            keepGoing = false;
+                            continue;
+                        }
+                        // fall-through
+                    default:
+                        setNext(element->next);
+                        keepGoing = true;
+                        break;
+                }
+                break;
+            case FLOW_ENTRY: keepGoing = handleEntry(); break;
+            case FLOW_JUMP:  keepGoing = handleJump(); break;
+        }
+    }
+    return 0;
+}
+
+static const s32 sTargetActorIds[] = {0,  1,  10, 46, 47, 48, 49, 50, 51, 52, 53,
+                                      54, 55, 65, 67, 77, 81, 87, 88, 89, 90, -1};
```

Comment:
what actor ids do these represent?

Reply:
Good question. Looking once more into the files I realized that this controls whether Fi will give you a kill count for a given actor or not, but I don't know where these IDs are defined (perhaps EnData?)

Reply:
Nothing conclusive but I added my insight about the purpose of this branch.